pandas/tests/io/test_excel.py

```diff
@@ -511,7 +511,7 @@ def test_read_one_empty_col_with_header(self):
             )
         expected_header_none = DataFrame(pd.Series([0], dtype='int64'))
         tm.assert_frame_equal(actual_header_none, expected_header_none)
-        expected_header_zero = DataFrame(columns=[0], dtype='int64')
+        expected_header_zero = DataFrame(columns=[0])
```

Comment:
why is this changing?

Reply:
The test was wrong and worked by accident. The result is, and should be, of ``object`` dtype; but the "expected" one was too, because the passed dtype wasn't being considered (see above).

Reply:
ok again add this as an example in a new issue

Reply:
Again #19646